### PR TITLE
fix(web): resolve release runtime warnings

### PR DESCRIPTION
**Pull request by hiroppy**
# Summary

- provide a stable production metadata base so social images never fall back to localhost
- initialize Recharts containers with positive dimensions to avoid pre-measurement warnings
- preserve the logo's intrinsic aspect ratio while keeping its rendered width unchanged
- add metadata unit coverage and an end-to-end regression check for all three warnings

## Linear Issue

- [HIR-109](https://linear.app/hiroppy/issue/HIR-109/new-release向けにmf-dashboardのエラーを修正)

### Selected Quality Characteristics

| Quality characteristic | Why it is relevant | Acceptance criteria |
| ---------------------- | ------------------ | ------------------- |
| Functional suitability | Metadata and dashboard rendering must produce correct runtime output | Social image URLs use the public origin and targeted warnings are absent |
| Compatibility | Recharts and Next Image behavior depends on browser layout and framework runtime | Desktop and mobile browser projects render existing charts and the header without regressions |
| Interaction capability | Warnings should not appear while users navigate primary dashboard pages | Primary page flows remain usable and warning-free |

### Test Techniques

| Test technique | Selection rationale | Expected coverage |
| -------------- | ------------------- | ----------------- |
| Boundary value analysis | Recharts warns before measurement at its -1 dimension boundary | Positive initial dimensions and measured dimensions are exercised |
| Equivalence partitioning | Primary pages cover the shared header and chart variants | Dashboard, cash flow, balance sheet, insights, and simulator routes |
| Error guessing | The ticket provides exact framework warning signatures | Browser console is filtered for all three regression messages |

### Primary Test Conditions

- No deployment URL variables: metadata falls back to the documented public demo origin.
- Explicit site and Vercel URLs: precedence and HTTPS normalization remain correct.
- Initial and measured chart layout: no non-positive dimension warning.
- Header logo: intrinsic 758x708 ratio renders at 32x29.875 without aspect-ratio warnings.
- Desktop Chrome, mobile Chrome, and mobile Safari application flows remain green.

### Out-of-Scope Risks

- Deployment-specific environment values are not exercised against a live Vercel deployment; URL precedence is covered by unit tests.

## Verification

- [x] Unit tests
- [x] Type checking
- [x] Lint
- [x] Storybook tests
- [x] E2E tests
- [x] Manual verification

### Commands Executed

```text
pnpm --filter @mf-dashboard/web test:unit — 24 files / 378 tests passed
pnpm turbo typecheck — 9 packages passed
pnpm lint — passed
pnpm --filter @mf-dashboard/web test:storybook — 74 files / 330 tests passed
pnpm --filter @mf-dashboard/web exec playwright test --reporter=dot — 45 tests passed
pnpm format:check — passed
Chromium runtime walkthrough on /, /cf, /bs, /insights, /accounts, /simulator — 0 targeted warnings; OG image resolved to https://mf-dashboard-demo.vercel.app/logo.png
```

### Checks Not Run

- N/A — all relevant checks were run

### CODE REVIEW

File: apps/web/src/lib/metadata.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const DEFAULT_SITE_URL = "https://mf-dashboard-demo.vercel.app";
+
+export function createMetadataBase(
+  environment: Record<string, string | undefined> = process.env,
+): URL {
+  const siteUrl =
+    environment.NEXT_PUBLIC_SITE_URL ??
+    environment.VERCEL_PROJECT_PRODUCTION_URL ??
+    environment.VERCEL_URL ??
+    DEFAULT_SITE_URL;
```
Comment (hiroppy) [inline apps/web/src/lib/metadata.ts]: This fallback makes deployments without `NEXT_PUBLIC_SITE_URL`/Vercel envs advertise `mf-dashboard-demo.vercel.app` as their metadata origin. The documented Docker/Cloudflare self-hosted setup does not set those variables, so Open Graph/Twitter image URLs will point to the public demo instead of the actual dashboard host. Please derive/configure the self-hosted origin (or document/set a required site URL) rather than defaulting production metadata to the demo domain.

Comment (hiroppy) [reply apps/web/src/lib/metadata.ts]: Addressed in 6e8e56c. Self-hosted production now requires the existing DASHBOARD_URL setting, passes it to both the Docker build and web runtime, and uses it for metadata. Demo, local development, and Vercel keep explicit environment-specific origins; production without a configured origin fails fast instead of advertising the demo domain.
